mne/source_estimate.py

```diff
@@ -2460,238 +2265,19 @@ def morph_data(subject_from, subject_to, stc_from, grade=5, smooth=None,
-        and 'grade=5', this set of vertices will automatically be used
-        (instead of computed) for speed, since this is a common morph.
-    subjects_dir : string, or None
-        Path to SUBJECTS_DIR if it is not set in the environment
-    n_jobs : int
-        Number of jobs to run in parallel
-    verbose : bool, str, int, or None
-        If not None, override default verbose level (see :func:`mne.verbose`
-        and :ref:`Logging documentation <tut_logging>` for more).
-
-    Returns
-    -------
-    vertices : list of arrays of int
-        Vertex numbers for LH and RH
-    """
-    # add special case for fsaverage for speed
-    if subject == 'fsaverage' and grade == 5:
-        return [np.arange(10242), np.arange(10242)]
-    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
-
-    spheres_to = [op.join(subjects_dir, subject, 'surf',
-                          xh + '.sphere.reg') for xh in ['lh', 'rh']]
-    lhs, rhs = [read_surface(s)[0] for s in spheres_to]
-
-    if grade is not None:  # fill a subset of vertices
-        if isinstance(grade, list):
-            if not len(grade) == 2:
-                raise ValueError('grade as a list must have two elements '
-                                 '(arrays of output vertices)')
-            vertices = grade
-        else:
-            # find which vertices to use in "to mesh"
-            ico = _get_ico_tris(grade, return_surf=True)
-            lhs /= np.sqrt(np.sum(lhs ** 2, axis=1))[:, None]
-            rhs /= np.sqrt(np.sum(rhs ** 2, axis=1))[:, None]
-
-            # Compute nearest vertices in high dim mesh
-            parallel, my_compute_nearest, _ = \
-                parallel_func(_compute_nearest, n_jobs)
-            lhs, rhs, rr = [a.astype(np.float32)
-                            for a in [lhs, rhs, ico['rr']]]
-            vertices = parallel(my_compute_nearest(xhs, rr)
-                                for xhs in [lhs, rhs])
-            # Make sure the vertices are ordered
-            vertices = [np.sort(verts) for verts in vertices]
-            for verts in vertices:
-                if (np.diff(verts) == 0).any():
-                    raise ValueError(
-                        'Cannot use icosahedral grade %s with subject %s, '
-                        'mapping %s vertices onto the high-resolution mesh '
-                        'yields repeated vertices, use a lower grade or a '
-                        'list of vertices from an existing source space'
-                        % (grade, subject, len(verts)))
-    else:  # potentially fill the surface
-        vertices = [np.arange(lhs.shape[0]), np.arange(rhs.shape[0])]
-
-    return vertices
+    return SourceMorph(subject_from=subject_from,
+                       subject_to=subject_to, spacing=grade,
+                       smooth=smooth, subjects_dir=subjects_dir,
+                       warn=warn, verbose=verbose)(stc_from)
```

Comment:
@TommyClausner like you did here stc.morph should internally use a SourceMorph object.

Reply:
yeah but here it's the "actual" function - before it was similar: the method called this function - so I kept the structure rather than wrapping it twice